### PR TITLE
[13.0] sale_by_packaging: no pkging override if already set

### PR DESCRIPTION
**Pull request by simahawk**
Before this change the packaging was always overridden
hence the user had not control on the packaging to be used on SO line.

TODO:

- [x] rewrite tests

### CODE REVIEW

File: sale_by_packaging/models/sale_order_line.py [hunk 1]
```diff
@@ -101,14 +101,22 @@ def _get_product_packaging_having_multiple_qty(self, product, qty, uom):
...
-        if vals.get("product_packaging") or not any(
-            fname in vals for fname in fields_to_check
-        ):
+        if vals.get("product_packaging") or self.env.context.get("_skip_auto_assign"):
+            # setting the packaging directly, skip auto assign
             return super().write(vals)
         for line in self:
+            if line.product_packaging:
+                # don't touch it if set already
+                continue
```
Comment (TDu) [inline sale_by_packaging/models/sale_order_line.py:114]: This is problematic in my view, I open a quotation change the quantity on a line, click save, there is no message and the changes are not saved

**Review by jgrandguillaume — APPROVED** [review]
LGTM

File: sale_by_packaging/models/sale_order_line.py (hunk 1 above)
Comment (sebalix) [reply sale_by_packaging/models/sale_order_line.py:114]: fixed

**Review by TDu — APPROVED** [review]

**Comment by TDu** [comment]
@simahawk  Looks good as it is now thanks to @sebalix for the last fix. Does it still need the `need fixing` and `work in progress` labels ?

**Comment by simahawk** [comment]
fixes integrated into https://github.com/OCA/sale-workflow/pull/1276

**Pull request closed without merging**